node/heartbeat/monitor.go

```diff
@@ -64,6 +73,8 @@ func NewMonitor(
 	mon := &Monitor{
 		marshalizer:                 marshalizer,
 		heartbeatMessages:           make(map[string]*heartbeatMessageInfo),
+		eligibleListProvider:        eligibleListProvider,
+		epochHandler:                epochHandler,
```

Comment:
epochHandler is not testes for nil. Is it ok?

Reply:
I will add a check for it but we should rethink these things. In this situation let's say it is ok because epochhandler comes from a `with` option. And for example if you forget that with in an integration test, it should fail. But there are cases where we forward a component many layers down and check it for nil every time.

Reply:
fixed on line 71 now